$NetBSD: patch-ac,v 1.3 2001/07/18 20:28:35 dmcmahill Exp $

work around a compiler bug which causes arrowheads to not
display correctly on NetBSD-1.5 and older alpha's when 
optimization is used.

--- poly.c.orig	Sat Aug  5 09:57:25 2000
+++ poly.c	Wed Jul 18 16:12:24 2001
@@ -3232,4 +3232,5 @@
    double len, sin, cos;
    XGCValues values;
+   short tmps;
 
    trans_pat = ObjPtr->trans_pat;
@@ -3320,8 +3321,8 @@
       tmp_v[0].x = tmp_v[3].x = v0.x;
       tmp_v[0].y = tmp_v[3].y = v0.y;
-      tmp_v[1].x = round(v0.x + aw*cos - ah*sin);
-      tmp_v[1].y = round(v0.y + aw*sin + ah*cos);
-      tmp_v[2].x = round(v0.x + aw*cos + ah*sin);
-      tmp_v[2].y = round(v0.y + aw*sin - ah*cos);
+      tmps = round(v0.x + aw*cos - ah*sin); tmp_v[1].x = tmps;
+      tmps = round(v0.y + aw*sin + ah*cos); tmp_v[1].y = tmps;
+      tmps = round(v0.x + aw*cos + ah*sin); tmp_v[2].x = tmps; 
+      tmps = round(v0.y + aw*sin - ah*cos); tmp_v[2].y = tmps;
 
       XFillPolygon(mainDisplay, Win, drawGC, tmp_v, 4, Convex,
@@ -3340,8 +3341,8 @@
       tmp_v[0].x = tmp_v[3].x = vnminus1.x;
       tmp_v[0].y = tmp_v[3].y = vnminus1.y;
-      tmp_v[1].x = round(vnminus1.x - aw*cos + ah*sin);
-      tmp_v[1].y = round(vnminus1.y - aw*sin - ah*cos);
-      tmp_v[2].x = round(vnminus1.x - aw*cos - ah*sin);
-      tmp_v[2].y = round(vnminus1.y - aw*sin + ah*cos);
+      tmps = round(vnminus1.x - aw*cos + ah*sin); tmp_v[1].x = tmps;
+      tmps = round(vnminus1.y - aw*sin - ah*cos); tmp_v[1].y = tmps;
+      tmps = round(vnminus1.x - aw*cos - ah*sin); tmp_v[2].x = tmps;
+      tmps = round(vnminus1.y - aw*sin + ah*cos); tmp_v[2].y = tmps;
 
       XFillPolygon(mainDisplay, Win, drawGC, tmp_v, 4, Convex,
